### ISSUE TITLE
sist2 web UI localization

### ISSUE THREAD

**Issue by simon987** [sist2app/sist2#180]
Reply to this issue if you would like to help me add more languages for the sist2 web UI!

The currently supported languages are:

- `en` English
- `fr` French
- `zh-CN` Simplified Chinese
- `de` German
- `pl` Polish

**Comment by panekj** [comment]
I can help

**Comment by simon987** [comment]
@panekj Great!

Could you create a new language in https://github.com/simon987/sist2/blob/master/sist2-vue/src/i18n/messages.ts ?. You can use `pl` for Polish. The message names and correspondence in English should be pretty self-explanatory but let me know if there are any ambiguities!

**Comment by yatli** [comment]
I can do zh-CN.

**Comment by simon987** [comment]
> I can do zh-CN.

That would be awesome!

If you want to test the changes in the web app, you can run `sist2 web` with the `--dev` argument in the root folder of the repository while `sist2-vue/watch.sh` (after running `npm install` inside `sist2-vue`) is running and it should update without needing to (re)compile the C binary. Let me know if you have any problems with that

**Comment by yatli** [comment]
ooh, that's good to know!

**Comment by simon987** [comment]
> npm install not working due to fslightbox-vue dep path hard-coded

Oh, right... I'm using the fslightbox-vue pro version. I don't really know any good alternatives that's why I'm using it. For now if you want to compile the web UI send me your email I can just send you the link to the tarball

**Comment by yatli** [comment]
Thanks, ./watch.sh with human in the loop would be too obtrusive for you tho :D

Guess I can just remove it from package.json and remove the lightbox component...

**Comment by simon987** [comment]
Yeah I forgot about this library. In the next version I will try to use NPM's `optionalDependencies` thing and gracefully disable the lightbox component when it detects that the library is not installed. Ideally I should move away from fslightbox altogether but I really haven't found a good alternative that works with mobile clients yet

**Comment by yatli** [comment]
Actually just remove the pro version and then `npm install --save fslightbox-vue` and it compiles fine. You'll only need to compile it once someone's code is checked in.

**Comment by MarioSob** [comment]
@simon987 Is the topic of Polish translation dead? I can't see it in the actual application.
I can provide such translation if you still don't have it, just let me know.

BTW: I would really like to see OCR language pre-installed in default Docker image for Polish as well.

**Comment by MarioSob** [comment]
I have no rights to push my custom branch with changes to your repo: `fatal: unable to access 'https://github.com/simon987/sist2.git/': The requested URL returned error: 403`

**Comment by simon987** [comment]
That's okay, you can create a fork under your account and try again with `https://github.com/MarioSob/sist2` instead

If that doesn't work you can also just email me the modified messages.ts file at `me@simon987.net` or paste them here

**Comment by MarioSob** [comment]
I've send you an email.

**Comment by simon987** [comment]
Much appreciated @MarioSob. I added the changes in https://github.com/simon987/sist2/commit/41ef94062347f5394f62d675987247d31b3923a4
